bindings/sftp/sftp.go

```diff
@@ -0,0 +1,314 @@
+	if err != nil {
+		return nil, err
+	}
+
+	return &m, nil
+}
+
+func (sftp *Sftp) Operations() []bindings.OperationKind {
+	return []bindings.OperationKind{
+		bindings.CreateOperation,
+		bindings.GetOperation,
+		bindings.DeleteOperation,
+		bindings.ListOperation,
+	}
+}
+
+func (sftp *Sftp) create(_ context.Context, req *bindings.InvokeRequest) (*bindings.InvokeResponse, error) {
+	metadata, err := sftp.metadata.mergeWithRequestMetadata(req)
+	if err != nil {
+		return nil, fmt.Errorf("sftp binding error: error merging metadata: %w", err)
+	}
+
+	path, err := metadata.getPath()
+	if err != nil {
+		return nil, fmt.Errorf("sftp binding error: %w", err)
+	}
+
+	dir, fileName := sftpClient.Split(path)
+
+	err = sftp.sftpClient.MkdirAll(dir)
+	if err != nil {
+		return nil, fmt.Errorf("sftp binding error: error create dir: %s %w", dir, err)
+	}
+
+	file, err := sftp.sftpClient.Create(path)
+	if err != nil {
+		return nil, fmt.Errorf("sftp binding error: error create file: %s %w", path, err)
+	}
+
+	_, err = file.Write(req.Data)
+	if err != nil {
+		return nil, fmt.Errorf("sftp binding error: error write file: %w", err)
+	}
+
+	jsonResponse, err := json.Marshal(createResponse{
+		FileName: fileName,
+	})
+	if err != nil {
+		return nil, fmt.Errorf("sftp binding error: error marshalling create response: %w", err)
+	}
+
+	return &bindings.InvokeResponse{
+		Data: jsonResponse,
+		Metadata: map[string]string{
+			metadataFileName: fileName,
+		},
+	}, nil
+}
+
+func (sftp *Sftp) list(_ context.Context, req *bindings.InvokeRequest) (*bindings.InvokeResponse, error) {
+	metadata, err := sftp.metadata.mergeWithRequestMetadata(req)
+	if err != nil {
+		return nil, fmt.Errorf("sftp binding error: error merging metadata: %w", err)
+	}
+
+	path, err := metadata.getPath()
+	if err != nil {
+		return nil, fmt.Errorf("sftp binding error: %w", err)
+	}
+
+	files, err := sftp.sftpClient.ReadDir(path)
+	if err != nil {
+		return nil, fmt.Errorf("sftp binding error: error read dir: %s %w", path, err)
+	}
+
+	var resp []listResponse
+
+	for _, file := range files {
+		resp = append(resp, listResponse{
+			FileName:    file.Name(),
+			IsDirectory: file.IsDir(),
+		})
+	}
+
+	jsonResponse, err := json.Marshal(resp)
+	if err != nil {
+		return nil, fmt.Errorf("sftp binding error: cannot marshal list to json: %w", err)
+	}
+
+	return &bindings.InvokeResponse{
+		Data: jsonResponse,
+	}, nil
+}
+
+func (sftp *Sftp) get(_ context.Context, req *bindings.InvokeRequest) (*bindings.InvokeResponse, error) {
+	metadata, err := sftp.metadata.mergeWithRequestMetadata(req)
+	if err != nil {
+		return nil, fmt.Errorf("sftp binding error: error merging metadata: %w", err)
+	}
+
+	path, err := metadata.getPath()
+	if err != nil {
+		return nil, fmt.Errorf("sftp binding error: %w", err)
+	}
+
+	file, err := sftp.sftpClient.Open(path)
+	if err != nil {
+		return nil, fmt.Errorf("sftp binding error: error open file: %s %w", path, err)
```

Comment:
Please do the same on the other ones:
```suggestion
		return nil, fmt.Errorf("sftp binding error: error open file %s: %w", path, err)
```